apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx

```diff
@@ -0,0 +1,558 @@
+    IdPEntityId?: string;
+    LogoutReqUrl?: string;
+    NameIDType?: string;
+    RequestMethod?: string;
+    SPEntityId?: string;
+    SSOUrl?: string;
+    SignatureAlgorithm?: string;
+    commonAuthQueryParams?: string;
+    ISAuthnReqSigned?: boolean;
+    IncludeProtocolBinding?: boolean;
+    IsAuthnRespSigned?: boolean;
+    IsLogoutEnabled?: boolean;
+    IsLogoutReqSigned?: boolean;
+    IsSLORequestAccepted?: boolean;
+    IsUserIdInClaims?: boolean;
+}
+
+export const SamlAuthenticatorSettingsForm: FunctionComponent<SamlSettingsFormProps> = (
+    props: SamlSettingsFormProps
+): ReactElement => {
+
+    const {
+        authenticator,
+        onSubmit,
+        ["data-testid"]: testId
+    } = props;
+
+    const config: ConfigReducerStateInterface = useSelector((state: AppState) => state.config);
+    const { t } = useTranslation();
+
+    const initialFormValues = useMemo<SamlProperties>(() => {
+
+        const [findPropVal] = fastSearch(authenticator);
+
+        return {
+            DigestAlgorithm: findPropVal<string>({ defaultValue: "SHA1", key: "DigestAlgorithm" }),
+            IdPEntityId: findPropVal<string>({ defaultValue: "", key: "IdPEntityId" }),
+            NameIDType: findPropVal<string>({ defaultValue: "", key: "NameIDType" }),
+            RequestMethod: findPropVal<string>({ defaultValue: "", key: "RequestMethod" }),
+            SPEntityId: findPropVal<string>({ defaultValue: "", key: "SPEntityId" }),
+            SSOUrl: findPropVal<string>({ defaultValue: "", key: "SSOUrl" }),
+            SignatureAlgorithm: findPropVal<string>({ defaultValue: "RSA with SHA1", key: "SignatureAlgorithm" }),
+            commonAuthQueryParams: findPropVal<string>({ defaultValue: "", key: "commonAuthQueryParams" }),
+            LogoutReqUrl: findPropVal<string>({ defaultValue: "", key: "LogoutReqUrl" }),
+            IncludeProtocolBinding: findPropVal<boolean>({ defaultValue: false, key: "IncludeProtocolBinding" }),
+            ISAuthnReqSigned: findPropVal<boolean>({ defaultValue: false, key: "ISAuthnReqSigned" }),
+            // `IsAuthnRespSigned` is by default set to true when creating the SAML IdP so,
+            // always the value will be true. Keeping this here to indicate for the user and
+            // to enable this if requirements gets changed.
+            IsAuthnRespSigned: findPropVal<boolean>({ defaultValue: true, key: "IsAuthnRespSigned" }),
+            IsLogoutEnabled: findPropVal<boolean>({ defaultValue: false, key: "IsLogoutEnabled" }),
+            IsLogoutReqSigned: findPropVal<boolean>({ defaultValue: false, key: "IsLogoutReqSigned" }),
+            IsSLORequestAccepted: findPropVal<boolean>({ defaultValue: true, key: "IsSLORequestAccepted" }),
+            IsUserIdInClaims: findPropVal<boolean>({ defaultValue: true, key: "IsUserIdInClaims" })
+        } as SamlProperties;
+
+    }, []);
+
+    const [formValues, setFormValues] = useState<SamlProperties>({} as SamlProperties);
```

Comment:
Lets move the `useState` hooks to the top of the component for readablity.

**Order**

1. Props destructing.
2. Custom hooks (redux, i18n)
3. Redux selectors.
4. Use State Hooks.
5. Use Effect (Initial mount first).
6. Use Memo.
7. Use Callback.
8. Functions.
9. Return.

Reply:
Was able to partially address this. The `useMemo` declaration should be first because otherwise `useEffect` complaints it's used before declaration. Since we use `const` we cant declare and then initialize after `useEffect` or to switch the order right?